src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -138,20 +146,25 @@ public boolean modifySubscription(Subscription subscription, String subscription
     @Override
     public boolean deleteSubscription(String subscriptionName) throws AccessException {
         String ldapUserName = (ldapEnabled) ? HttpSessionConfig.getCurrentUser() : "";
-        String deleteQuery = generateQuery(subscriptionName, ldapUserName);
+        String deleteQuery;
+        boolean ownerExist = doSubscriptionOwnerExist(subscriptionName);
+        if (ownerExist) {
```

Comment:
This if-statement is duplicated, I would prefere if the if-statement was in the generateQuery function.